shared/util/pictures.js

```diff
@@ -9,128 +9,143 @@ type Info = {
+      const cbs = info.callbacks
       info.callbacks = []
+      cbs.forEach(cb => cb(username, urlMap))
     }
   })
 
-  if (!good.length) {
-    return
-  }
-
-  const [teamnames, usernames] = partition(good, g => _nameToURL[g].isTeam)
-
-  if (usernames.length) {
-    apiserverGetRpcPromise({
-      param: {
-        args: [
-          {key: 'usernames', value: good.join(',')},
-          {key: 'formats', value: 'square_360,square_200,square_40'},
-        ],
-        endpoint: 'image/username_pic_lookups',
-      },
-    })
-      .then(response => {
-        JSON.parse(response.body).pictures.forEach((picMap, idx) => {
-          const username = usernames[idx]
-          let urlMap = {
-            ...(picMap['square_200'] ? {'200': picMap['square_200']} : null),
-            ...(picMap['square_360'] ? {'360': picMap['square_360']} : null),
-            ...(picMap['square_40'] ? {'40': picMap['square_40']} : null),
-          }
-
-          const info = _nameToURL[username]
-          if (info) {
-            info.done = true
-            info.urlMap = urlMap
-            info.callbacks.forEach(cb => cb(username, urlMap))
-            info.callbacks = []
-          }
-        })
-      })
-      .catch(() => {
-        usernames.forEach(username => {
-          const info = _nameToURL[username]
-          const urlMap = {}
-          if (info) {
-            info.done = true
-            info.error = true
-            info.callbacks.forEach(cb => cb(username, urlMap))
-            info.callbacks = []
-          }
-        })
-      })
-  }
-
-  if (teamnames.length) {
-    apiserverGetRpcPromise({
-      param: {
-        args: [
-          {key: 'team_names', value: good.join(',')},
-          {key: 'formats', value: 'square_360,square_200,square_40'},
-        ],
-        endpoint: 'image/team_avatar_lookups',
-      },
-    })
-      .then(response => {
-        JSON.parse(response.body).pictures.forEach((picMap, idx) => {
-          const teamname = teamnames[idx]
-          let urlMap = {
-            ...(picMap['square_200'] ? {'200': picMap['square_200']} : null),
-            ...(picMap['square_360'] ? {'360': picMap['square_360']} : null),
-            ...(picMap['square_40'] ? {'40': picMap['square_40']} : null),
-          }
-
-          const info = _nameToURL[teamname]
-          if (info) {
-            info.done = true
-            info.urlMap = urlMap
-            info.callbacks.forEach(cb => cb(teamname, urlMap))
-            info.callbacks = []
-          }
-        })
-      })
-      .catch(() => {
-        teamnames.forEach(teamname => {
-          const info = _nameToURL[teamname]
-          const urlMap = {}
-          if (info) {
-            info.done = true
-            info.error = true
-            info.callbacks.forEach(cb => cb(teamname, urlMap))
-            info.callbacks = []
-          }
-        })
-      })
-  }
+  const [teamnames, usernames] = partition(good || [], n => _nameToInfo[n].isTeam)
```

Comment:
does partition ever return non-arrays? is `|| []` needed?

Reply:
this is us passing to partition and its not needed, removed, tnx